File: prqlc/bindings/php/src/Span.php
<?php

declare(strict_types=1);

namespace Prql\Compiler;

/**
 * Identifier of a location in source.
 * Contains offsets in terms of chars.
 */
final class Span
{
    /**
     * Start offset.
     */
    public int $start;

    /**
     * End offset.
     */
    public int $end;
}

File: prqlc/bindings/dotnet/PrqlCompiler/Span.cs
namespace Prql.Compiler;

/// <summary>
/// Identifier of a location in source.
/// Contains offsets in terms of chars.
/// </summary>
/// <param name="Start">Start offset.</param>
/// <param name="End">End offset.</param>
public readonly record struct Span(ulong Start, ulong End);
